sdk/ai/Azure.AI.Projects/src/Custom/Agent/AzureFunctionToolDefinition.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using Azure.Core;
+
+namespace Azure.AI.Projects
+{
+    [CodeGenSuppress("AzureFunctionToolDefinition", typeof(InternalAzureFunctionDefinition))]
+    public partial class AzureFunctionToolDefinition
```

Comment:
Why do we need to add this custom class?

Reply:
Similar to [FunctionToolDefinition](https://github.com/Azure/azure-sdk-for-net/blob/45adadefa2d39d2884f0889a240a8a7fbf096686/sdk/ai/Azure.AI.Projects/src/Custom/Agent/FunctionToolDefinition.cs#L22) we need to hide the internal constructor of FunctionDefinition, wich is marked [as internal for C# projects](https://github.com/Azure/azure-rest-api-specs/blob/f852701302da0c3b929f844982d4daa8b0a0999c/specification/ai/Azure.AI.Projects/agents/client.tsp#L37C3-L37C49). It also helps to avoid complexity of multiple data class creation, which, in turn depends on JSON, taken by server.

Reply:
Ahh, I understand. There are some issues in the existing spec, so we shouldn't follow the existing approach if we believe we can improve it. Instead, we should implement what we think is the best solution.  

If the `InternalAzureFunctionDefinition` type is not being used anywhere, I would suggest removing it from the `TypeSpec` and updating the `AzureFunctionToolDefinition` type to include `Name`, `Description`, and `Parameters` properties directly in it.

Reply:
The class hierarchy should follow the specification, so that we will get the correct JSON when we call the service. Essentially, each dictionary is translated to object. Here is the example of a JSON, we want to get:
```JSON
{
    // Top level AzureFunctionToolDefinition
    "type": "azure_function", 
    "azure_function": {
        // InternalAzureFunctionDefinition
        "function": {
           // InternalFunctionDefinition
            "name": "foo",
            "description": "Get answers from the foo bot.",
            "parameters": {
                "type": "object",
                "properties": {
                    "query": {"type": "string", "description": "The question to ask."},
                    "outputqueueuri": {"type": "string", "description": "The full output queue uri."},
                },
                "required": ["query"],
            },
        },
        "input_binding": {
            // AzureStorageQueueBinding
            "type": "storage_queue",
            "storage_queue": {
                // AzureFunctionStorageQueue
                "queue_service_uri": "https://xxxxxxx.queue.core.windows.net",
                "queue_name": "azure-function-foo-input",
            },
        },
        "output_binding": {
            // AzureStorageQueueBinding
            "type": "storage_queue",
            "storage_queue": {
                // AzureFunctionStorageQueue
                "queue_service_uri": "https://xxxxxxx.queue.core.windows.net",
                "queue_name": "azure-function-tool-output",
            },
        },
    },
}
```
Should we keep InternalAzureFunctionDefinition internal?